pgmpy/readwrite/XMLBIF.py

```diff
@@ -233,20 +242,24 @@ def __str__(self):
         """
         if self.prettyprint:
             self.indent(self.xml)
-        return etree.tostring(self.xml, encoding=self.encoding)
+        f = BytesIO()
+        et = etree.ElementTree(self.xml)
+        et.write(f, encoding='utf-8', xml_declaration=True)
+        return f.getvalue().decode('utf-8')  # your XML file, encoded as UTF-8
+        # return etree.tostring(self.xml, encoding=self.encoding)
```

Comment:
previous behaviour was returning bytes from __str__ (meaning calling print on the writer resulted in an error). It was also not outputting a header - which is required for some other programs to be able to read ie JavaBayes.

I'll fix to use encoding parameter